runtime/lua/vim/lsp/buf.lua

```diff
@@ -296,11 +296,23 @@ function M.rename(new_name)
       --
       -- see https://microsoft.github.io/language-server-protocol/specification#textDocument_prepareRename
       opts.default = vfn.expand('<cword>')
-      if not new_name then npcall(vim.ui.input, opts, on_confirm) end
+      if not new_name then npcall(vim.ui.input, opts, request_rename) end
     end
-    if new_name then on_confirm(new_name) end
+    if new_name then request_rename(new_name) end
+  end
+
+  -- Only send prepareRename request if all servers attached to the buffer support it
+  local supports_prepare_rename = true
+  for _, client in pairs(vim.lsp.buf_get_clients()) do
+    supports_prepare_rename = client.supports_method('textDocument/prepareRename')
+  end
```

Comment:
Hmm, I haven't kept up with the recent changes, but wasn't it the opposite originally? We'd send prepareRename if at least one of the servers supported it, and then issue the rename on the callback for the specific client? Otherwise prepareRename will basically never work with multiple servers? (Sample setup: efm for formatting + full-fledged server that supports prepareRename for rename, in that case prepareRename would never get called?).

On a side note, the comment doesn't match the code 🙈  the code will take whatever the last client in the list says (perhaps you're missing an `and` or a conditional `break`?).

Reply:
No we didn't have prepareRename and the capability was missing. I'm ok with requiring all servers must support prepareRename because that's a limitation with our current request system, and I'll fix the code, it should be supports_prepare rename = supports_rename and ...

Reply:
Gotcha, thanks for clarifying!